networkx/generators/tests/test_directed.py

```diff
@@ -64,6 +65,18 @@ def test_scale_free_graph_initial_graph_kwarg(ig):
         scale_free_graph(100, initial_graph=ig)
 
 
+@pytest.mark.parametrize("create_using", [None, MultiDiGraph()])
+@pytest.mark.parametrize("graph", ["gn", "gnr", "gnc"])
+def test_gn_graph_generators_single_node(create_using, graph):
+    if graph == "gn":
+        G = gn_graph(1, create_using=create_using, seed=1)
+    elif graph == "gnr":
+        G = gnr_graph(1, 0.5, create_using=create_using, seed=1)
+    elif graph == "gnc":
+        G = gnc_graph(1, create_using=create_using, seed=1)
+    assert G.number_of_nodes() == 1 and G.number_of_edges() == 0
+
+
```

Comment:
Very nice @chimaobi-okite . A couple more minor suggestions: one is that you can actually pass in the functions themselves to the parametrization, so there's no need for the extra logic mapping a string key to the function. Of course, I assume you did this because it turns out that the signature for `gnr_graph` differs from the other two, which makes parametrizing harder. Your solution is clever! An alternative organization might look something like:

```suggestion
@pytest.mark.parametrize("create_using", [None, MultiDiGraph])
@pytest.mark.parametrize("gn_func", (gn_graph, gnc_graph))
def test_gn_gnc_generators_single_node(create_using, gn_func):
    G = gn_func(1, create_using=create_using, seed=1)
    assert G.number_of_nodes() == 1 and G.number_of_edges() == 0
    
@pytest.mark.parametrize("create_using", [None, MultiDiGraph])
def test_gnr_graph_single_node(create_using):
    G = gnr_graph(1, 0.5, create_using=create_using, seed=1)
    assert G.number_of_nodes() == 1 and G.number_of_edges() == 0


```

Reply:
Thanks a lot, @rossbar, this looks better, I never thought of it this way.
Thanks once more, I have learned new things as regards code testing from you. The updates have been pushed.